File: java/Text.java
public class Text {
    private String content;

    public Text(String content) {
        this.content = content;
    }

    public AnalysisResult analyze() {
        int numWords = content.split("\\s+").length;
        int numChars = content.length();
        int numLines = content.split("\r\n|\r|\n").length;

        return new AnalysisResult(numWords, numChars, numLines);
    }

    public static class AnalysisResult {
        private final int wordCount;
        private final int charCount;
        private final int lineCount;

        public AnalysisResult(int wordCount, int charCount, int lineCount) {
            this.wordCount = wordCount;
            this.charCount = charCount;
            this.lineCount = lineCount;
        }

        public int getWordCount() {
            return wordCount;
        }

        public int getCharCount() {
            return charCount;
        }

        public int getLineCount() {
            return lineCount;
        }
    }

    public static void main(String[] args) {
        String textContent = "This is an example text.\nIt has multiple lines.\nAnd some characters.";

        Text textAnalyzer = new Text(textContent);
        AnalysisResult result = textAnalyzer.analyze();

        System.out.println("Word Count: " + result.getWordCount());
        System.out.println("Character Count: " + result.getCharCount());
        System.out.println("Line Count: " + result.getLineCount());
    }
}

File: c++/Text.cpp
#include <iostream>
#include <string>

class Text {
private:
    std::string content;

public:
    Text(std::string content) : content(content) {}

    struct AnalysisResult {
        int wordCount;
        int charCount;
        int lineCount;
    };

    AnalysisResult analyze() {
        int numWords = 0;
        int numChars = content.length();
        int numLines = 1;

        size_t pos = 0;
        while ((pos = content.find(' ', pos)) != std::string::npos) {
            ++numWords;
            ++pos;
        }
        ++numWords; // Count the last word

        pos = 0;
        while ((pos = content.find('\n', pos)) != std::string::npos) {
            ++numLines;
            ++pos;
        }

        AnalysisResult result = { numWords, numChars, numLines };
        return result;
    }
};

int main() {
    std::string textContent = "This is an example text.\nIt has multiple lines.\nAnd some characters.";

    Text textAnalyzer(textContent);
    Text::AnalysisResult result = textAnalyzer.analyze();

    std::cout << "Word Count: " << result.wordCount << std::endl;
    std::cout << "Character Count: " << result.charCount << std::endl;
    std::cout << "Line Count: " << result.lineCount << std::endl;

    return 0;
}
